src/compiler/utilities.ts

```diff
@@ -1713,16 +1713,20 @@ namespace ts {
 
     /**
      * Creates a deep clone of an EntityName, with new parent pointers.

```

Comment:
This looks like a left-deep clone. Or does `getMutableClone` do right-deep cloning somehow?